src/software/ai/hl/stp/tactic/passer/simulated_passer_tactic_test.cpp

```diff
@@ -0,0 +1,126 @@
+#include <gtest/gtest.h>
+
+#include <utility>
+
+#include "software/ai/hl/stp/tactic/move/move_tactic.h"
+#include "software/ai/hl/stp/tactic/passer/passer_tactic.h"
+#include "software/geom/algorithms/contains.h"
+#include "software/simulated_tests/simulated_tactic_test_fixture.h"
+#include "software/simulated_tests/terminating_validation_functions/ball_kicked_validation.h"
+#include "software/simulated_tests/terminating_validation_functions/robot_state_validation.h"
+#include "software/simulated_tests/validation/validation_function.h"
+#include "software/test_util/test_util.h"
+#include "software/time/duration.h"
+#include "software/world/world.h"
+
+class SimulatedPasserTacticTest
+    : public virtual SimulatedTacticTestFixture,
+      public ::testing::WithParamInterface<std::tuple<Pass, RobotStateWithId, BallState>>
+{
+};
+
+TEST_P(SimulatedPasserTacticTest, passer_test)
+{
+    Pass pass                    = std::get<0>(GetParam());
+    RobotStateWithId robot_state = std::get<1>(GetParam());
+    BallState ball_state         = std::get<2>(GetParam());
+
+    setBallState(ball_state);
+    addFriendlyRobots(TestUtil::createStationaryRobotStatesWithId({Point(-3, 2.5)}));
+    addFriendlyRobots({robot_state});
+
+    setRefereeCommand(RefereeCommand::NORMAL_START, RefereeCommand::FORCE_START);
+
+    auto tactic = std::make_shared<PasserTactic>(pass);
+    tactic->updateControlParams(pass);
+    setTactic(tactic);
+    setRobotId(1);
+
+    std::vector<ValidationFunction> terminating_validation_functions = {
+        [pass, tactic](std::shared_ptr<World> world_ptr,
+                       ValidationCoroutine::push_type& yield) {
+            ballKicked(pass.passerOrientation(), world_ptr, yield);
```

Comment:
Why are we checking for ball kicked first? Would be good to have some comments regardless

Reply:
yeah order is botched here, but it really doesn't matter too much cuz everything happens at once

re-ordered it to be more sane and added a comment